pkg/controller/build/build_controller.go

```diff
@@ -459,6 +468,25 @@ func (r *ReconcileBuild) validateBuildRunOwnerReferences(ctx context.Context, b
 	return nil
 }
 
+// validateSourceUrl returns error message if remote repository doesn't exist, otherwise, it will return empty string and nil
+func (r *ReconcileBuild) validateSourceUrl(ctx context.Context, b *build.Build, sourceUrl string) (string, error) {
+	prod, _ := zap.NewProduction()
+	logger := prod.Sugar()
+	switch b.GetAnnotations()[build.AnnotationBuildRefRepository] {
+	case "true":
+		if output, err := git.ListRemote(logger, sourceUrl); err != nil {
+			return output, err
+		} else {
+			return "", nil
+		}
```

Comment:
Since they share the same return signature, it can be simplified a bit I think.
```suggestion
		return git.ListRemote(logger, sourceUrl)
```

Reply:
Done!